### PR TITLE
Add debug headers when downloading plugins

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

This PR:

 - Adds the headers `grafana-retrycount` and `grafana-origin` to requests made to download plugins. The origin, since it's optional, is set in the context (I didn't want to add a parameter just for this). Let me know if you have a better proposal.
 - Refactors download logic to retry not only in case of panic but also if the server returns an error.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-community-team/issues/195

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by xnyo — APPROVED** [review]
LGTM! :raised_hands:

File: pkg/plugins/repo/client.go [hunk 1]
```diff
@@ -182,8 +196,10 @@ func (c *Client) sendReqNoTimeout(url *url.URL, compatOpts CompatOpts) (io.ReadC
 	return c.handleResp(res, compatOpts)
 }
 
-func (c *Client) createReq(url *url.URL, compatOpts CompatOpts) (*http.Request, error) {
-	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
+func (c *Client) createReq(ctx context.Context, url *url.URL, compatOpts CompatOpts) (*http.Request, error) {
+	u := url.String()
+
+	req, err := http.NewRequest(http.MethodGet, u, nil)
```
Comment (wbrowne) [inline pkg/plugins/repo/client.go:202]: Nit: Doesn't look like this needs to be split out?

File: pkg/plugins/repo/client.go [hunk 2]
```diff
@@ -145,8 +151,13 @@ func (c *Client) downloadFile(tmpFile *os.File, pluginURL, checksum string, comp
 
 	// Using no timeout as some plugin archives make take longer to fetch due to size, network performance, etc.
 	// Note: This is also used as part of the grafana plugin install CLI operation
-	bodyReader, err := c.sendReqNoTimeout(u, compatOpts)
+	bodyReader, err := c.sendReqNoTimeout(ctx, u, compatOpts)
 	if err != nil {
+		if c.retryCount < 3 {
+			c.retryCount++
+			c.log.Debug("Failed downloading. Will retry once.")
```
Comment (wbrowne) [inline pkg/plugins/repo/client.go:158]: I guess this log line (same as the original above) would be more correct to just say:
```suggestion
			c.log.Debug("Failed downloading. Will retry.")
```

**Review by wbrowne — APPROVED** [review]
Awesome 👏

File: pkg/plugins/repo/client.go (hunk 2 above)
Comment (andresmgot) [reply pkg/plugins/repo/client.go:158]: it will retry once... twice :D

**Review by wbrowne — APPROVED** [review]
🚀

**Pull request merged**